frontend/src/container/RoutingPolicies/__tests__/RoutingPolicies.test.tsx

```diff
@@ -0,0 +1,122 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import * as appHooks from 'providers/App/App';
+
+import RoutingPolicies from '../RoutingPolicies';
+import * as routingPoliciesHooks from '../useRoutingPolicies';
+import {
+	getAppContextMockState,
+	getUseRoutingPoliciesMockData,
+	MOCK_ROUTING_POLICY_1,
+} from './testUtils';
+
+jest.spyOn(appHooks, 'useAppContext').mockReturnValue(getAppContextMockState());
+
+jest.mock('../RoutingPolicyList', () => ({
+	__esModule: true,
+	default: jest.fn(() => (
+		<div data-testid="routing-policy-list">RoutingPolicyList</div>
+	)),
+}));
+jest.mock('../RoutingPolicyDetails', () => ({
+	__esModule: true,
+	default: jest.fn(() => (
+		<div data-testid="routing-policy-details">RoutingPolicyDetails</div>
+	)),
+}));
+jest.mock('../DeleteRoutingPolicy', () => ({
+	__esModule: true,
+	default: jest.fn(() => (
+		<div data-testid="delete-routing-policy">DeleteRoutingPolicy</div>
+	)),
+}));
+
+const mockHandleSearch = jest.fn();
+const mockHandlePolicyDetailsModalOpen = jest.fn();
+jest.spyOn(routingPoliciesHooks, 'default').mockReturnValue(
+	getUseRoutingPoliciesMockData({
+		setSearchTerm: mockHandleSearch,
+		handlePolicyDetailsModalOpen: mockHandlePolicyDetailsModalOpen,
+	}),
+);
+
+describe('RoutingPolicies', () => {
+	it('should render components properly', () => {
+		render(<RoutingPolicies />);
+		expect(screen.getByText('Routing Policies')).toBeInTheDocument();
+		expect(
+			screen.getByText('Create and manage routing policies.'),
+		).toBeInTheDocument();
+		expect(
+			screen.getByPlaceholderText('Search for a routing policy...'),
+		).toBeInTheDocument();
+		expect(
+			screen.getByRole('button', { name: /New routing policy/ }),
+		).toBeInTheDocument();
+		expect(screen.getByTestId('routing-policy-list')).toBeInTheDocument();
+		expect(
+			screen.queryByTestId('routing-policy-details'),
+		).not.toBeInTheDocument();
+		expect(screen.queryByTestId('delete-routing-policy')).not.toBeInTheDocument();
+	});
+
+	it('should enable the "New routing policy" button for users with ADMIN role', () => {
+		render(<RoutingPolicies />);
+		expect(
+			screen.getByRole('button', { name: /New routing policy/ }),
+		).toBeEnabled();
+	});
+
+	it('should disable the "New routing policy" button for users with NON_ADMIN roles', () => {
+		jest
+			.spyOn(appHooks, 'useAppContext')
+			.mockReturnValueOnce(getAppContextMockState({ role: 'VIEWER' }));
+		render(<RoutingPolicies />);
+		expect(
+			screen.getByRole('button', { name: /New routing policy/ }),
+		).toBeDisabled();
+	});
+
+	it('filters routing policies by search term', () => {
+		render(<RoutingPolicies />);
+		const searchInput = screen.getByPlaceholderText(
+			'Search for a routing policy...',
+		);
+		fireEvent.change(searchInput, {
+			target: { value: MOCK_ROUTING_POLICY_1.name },
+		});
+
+		expect(mockHandleSearch).toHaveBeenCalledWith(MOCK_ROUTING_POLICY_1.name);
```

Comment:
Hmm, do we need to mock the handle search, wouldn't it be better to not mock and ensure the handles search is indeed filtering them after the action?

Reply:
Same reason as the other one. The policy list is already mocked. So the fireEvent won't change it. But we can verify by making sure that the function that's supposed to handle the filtering is called with the correct arguments.